networkx/generators/tests/test_random_graphs.py

```diff
@@ -314,3 +291,39 @@ def root(u, w, r):
         graph = nx.random_kernel_graph(1000, integral, root)
         graph = nx.random_kernel_graph(1000, integral, root, seed=42)
         assert len(graph) == 1000
+
+
+@pytest.mark.parametrize(
+    ("k", "expected_num_nodes", "expected_num_edges"),
+    [
+        (2, 10, 10),
+        (4, 10, 20),
+    ],
+)
+def test_watts_strogatz(k, expected_num_nodes, expected_num_edges):
+    G = nx.watts_strogatz_graph(10, k, 0.25, seed=42)
+    assert len(G) == expected_num_nodes
+    assert G.number_of_edges() == expected_num_edges
+
+
+def test_newman_watts_strogatz_zero_probability():
+    G = nx.newman_watts_strogatz_graph(10, 2, 0.0, seed=42)
+    assert len(G) == 10
+    assert G.number_of_edges() == 10
+
+
+def test_newman_watts_strogatz_nonzero_probability():
+    G = nx.newman_watts_strogatz_graph(10, 4, 0.25, seed=42)
+    assert len(G) == 10
+    assert G.number_of_edges() >= 20
```

Comment:
Note that this `>=` is a directly copied over from the original test. I don't see any reason why it shouldn't be testing against the exact number of edges with this seed (e.g. `assert G.number_of_edges() == 31`), but I left this out for now since this PR is about reorganizing the tests rather than modifying them.

Reply:
Is it worth considering using `random.seed(42, version=2)` for forward compatibility? (https://docs.python.org/3/library/random.html#notes-on-reproducibility). I may be completely wrong about this, and the tests could always be changed in the future if a future version of Python breaks them...

Reply:
> Is it worth considering using random.seed(42, version=2) for forward compatibility? (https://docs.python.org/3/library/random.html#notes-on-reproducibility). I may be completely wrong about this, and the tests could always be changed in the future if a future version of Python breaks them...

Good question - this is something that's come up before as NumPy has introduced a new random number generator interface that will have similar concerns. We should definitely tackle the forward compatibility concerns for both `random` and `numpy.random` together. I suspect the best place to handle this would be in the `random_state` decorators.